src/main/scala/ai/privado/languageEngine/javascript/tagger/sink/JSAPITagger.scala

```diff
@@ -24,16 +24,52 @@
 package ai.privado.languageEngine.javascript.tagger.sink
 
 import ai.privado.cache.RuleCache
+import ai.privado.dataflow.DuplicateFlowProcessor
+import ai.privado.entrypoint.ScanProcessor
 import ai.privado.model.{Constants, RuleInfo}
 import ai.privado.tagger.sink.APITagger
+
+import scala.collection.mutable.ListBuffer
+import ai.privado.languageEngine.java.language.{NodeStarters, NodeToProperty, StepsForProperty}
+import ai.privado.languageEngine.java.semantic.JavaSemanticGenerator
 import ai.privado.utility.Utilities.{addRuleTags, getDomainFromTemplates, storeForTag}
 import io.shiftleft.codepropertygraph.generated.{Cpg, Operators}
-import io.shiftleft.semanticcpg.language._
+import io.shiftleft.semanticcpg.language.*
+import io.joern.dataflowengineoss.queryengine.{EngineConfig, EngineContext}
+import io.shiftleft.codepropertygraph.generated.nodes.{AstNode, CfgNode}
+import io.joern.dataflowengineoss.DefaultSemantics
+import ai.privado.utility.Utilities.{
+  addRuleTags,
+  getDomainFromString,
+  getFileNameForNode,
+  isFileProcessable,
+  storeForTag
+}
+import io.joern.dataflowengineoss.language.toExtendedCfgNode
+import overflowdb.BatchedUpdate
 
+// class GraphqlAPITagger
 class JSAPITagger(cpg: Cpg, ruleCache: RuleCache) extends APITagger(cpg, ruleCache) {
 
   override def runOnPart(builder: DiffGraphBuilder, ruleInfo: RuleInfo): Unit = {
     super.runOnPart(builder, ruleInfo)
+
+    // baseUrl" Identify the client creation & baseUrl used
+    // TODO: Replace the "baseUrl" with already tagged sinks
+    val ClientCreationBaseUrlPattern: String =
+      ruleCache.getSystemConfigByKey(Constants.clientCreationBaseUrlPattern, true)
+    val cacheCall          = cpg.call.where(_.nameNot("(<operator|<init).*")).l
+    val apiInternalSources = cpg.literal.code("(?:\"|'|`)(" + ruleInfo.combinedRulePattern + ")(?:\"|'|`)").l
+    val apis               = cacheCall.methodFullName(ClientCreationBaseUrlPattern).toList
```

Comment:
Looks to me like the `apiInternalSources`, `apis` these variables are part of the APITagger. Can we not override them?